src/with-theme.js

```diff
@@ -66,8 +91,11 @@ export default function withTheme(
         // as our `render` method without going through lifecycle hooks.
         // Also allows us to forward the context in the scenario where
         // a user wants to add more context.
-        // eslint-disable-next-line babel/new-cap
-        return ComponentToTheme({...this.props, ...this.state}, this.context)
+
+        /* eslint-disable max-len */
+        // $FlowFixMe: This may produce undesirable behavior when ComponentToTheme is a class instead of a function.
```

Comment:
That's what I did [here](https://github.com/paypal/glamorous/pull/205/files/6526214b9f057d26db60dd3856b75593c0afa0af#diff-3be2a7cdd58ade04e2acbcef3702d3d7R134). I don't think that fix would apply to this situation, unless I'm misunderstanding